File: Java-SpringBoot/BE-01-InMemory/java-21-springboot-4.0/src/main/java/com/flyrank/userservice/dto/ErrorResponse.java
package com.flyrank.userservice.dto;

import io.swagger.v3.oas.annotations.media.Schema;

/**
 * Standard error response payload matching FastAPI's {"detail": "..."} format.
 */
@Schema(description = "Standard error response payload")
public record ErrorResponse(
        @Schema(description = "Readable error message detail", example = "User with id '...' not found")
        String detail
) {}

File: Java-SpringBoot/BE-01-InMemory/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/userservice/dto/ErrorResponse.kt
package com.flyrank.userservice.dto

import io.swagger.v3.oas.annotations.media.Schema

/**
 * Standard error response payload matching FastAPI's {"detail": "..."} format.
 */
@Schema(description = "Standard error response payload")
data class ErrorResponse(
    @field:Schema(description = "Readable error message detail", example = "User with id '...' not found")
    val detail: String
)
